### ISSUE TITLE
Fix unhandled rejection in Chromium

### ISSUE THREAD

**Issue by micahjon** [jakearchibald/idb-keyval#93]
Over the past few year, we've been seeing a steady trickle of errors in production due to the following line.
https://github.com/jakearchibald/idb-keyval/blob/ea7d50752fd6a92eda003df021eb1976a0a35ef4/idb-keyval.ts#L12

Error: `Cannot read property 'createObjectStore' of undefined`

It only occurs in Chromium browsers (Chrome, Edge Chromium, Opera), and almost exclusively on Windows & Chrome OS. We haven't been able to replicate it, but it's happened to over 900 users by now.

Ideally it would just trigger a Promise rejection, but it doesn't appear to do that (in Sentry is shows up as an unhandled error even though we explicitly handle the Promise rejection).

Would it be worth checking for `openreq.result` before calling `createObjectStore()`? Thanks!

**Comment by jakearchibald** [comment]
Do you use a custom store name, or custom database name?

**Comment by micahjon** [comment]
Hmm, no, we just use the defaults:

```js
import * as idbKeyval from 'idb-keyval';

// ...

this.store = new idbKeyval.Store();
```

**Comment by jakearchibald** [comment]
Are you still seeing this? I can't really explain it from looking at the spec, and haven't been able to reproduce it locally.

**Comment by micahjon** [comment]
Yeah, unfortunately, it happens in production for us about 10-20 times a day (out of 100k users).

Screenshot from Sentry:
![image](https://user-images.githubusercontent.com/9473554/104235657-26113480-540a-11eb-84b1-36a455f33e90.png)

I looked at our logs and it appears that it only happens when a user switches tabs (from our tab to a different tab) while loading our app. So it's probably a side effect of Chrome reducing resources and throttling async tasks in background tabs.

As far as reproducing, it only occurs on Windows & Chrome OS--I don't see any bug reports for Mac.

**Comment by micahjon** [comment]
@jakearchibald , would it be worth opening a Chromium issue for this (assuming it conflicts with the IndexedDB spec)? I understand your reluctance to add more code to this library for something that occurs so rarely and appears to be a browser quirk.

**Comment by jakearchibald** [comment]
Go for it. Although it's difficult to do anything about it without steps to reproduce the issue.
